lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -2362,6 +2362,200 @@ class ConvertCastEquivalentOp : public OpConversionPattern<OpTy> {
+    SmallVector<Value> extractGridStride(gridRank, oneIndex);
+    int64_t lastGridDim = gridRank - 1;
+    Value twoIndex = rewriter.create<arith::ConstantIndexOp>(loc, 2);
+    extractGridShape[lastGridDim] = oneIndex;
+    extractGridStride[lastGridDim] = twoIndex;
+    SmallVector<Value> extractGridOffsets_1(gridRank, zeroIndex);
+    extractGridOffsets_1[lastGridDim] = oneIndex;
+    SmallVector<int64_t> gridShapeExtracted(gridShape);
+    gridShapeExtracted.back() = 1;
+    auto gridTypeExtracted =
+        RankedTensorType::get(gridShapeExtracted, gridElementType);
+    auto grid_0 = rewriter.create<tensor::ExtractSliceOp>(
+        loc, gridTypeExtracted, grid, extractGridOffsets_0, extractGridShape,
+        extractGridStride);
+    auto grid_1 = rewriter.create<tensor::ExtractSliceOp>(
+        loc, gridTypeExtracted, grid, extractGridOffsets_1, extractGridShape,
+        extractGridStride);
+    SmallVector<ReassociationIndices> associations;
+    associations.push_back(ReassociationIndices{0});
+    associations.push_back(ReassociationIndices{1});
+    associations.push_back(ReassociationIndices{2, 3});
+    SmallVector<int64_t> gridShapeCollapsed{gridShape[0], gridShape[1],
+                                            gridShape[2]};
+    gridShapeExtracted.back() = 1;
+    auto gridTypeCollapsed =
+        RankedTensorType::get(gridShapeCollapsed, gridElementType);
+    auto gridCollapsed_0 = rewriter.create<tensor::CollapseShapeOp>(
+        loc, gridTypeCollapsed, grid_0, associations);
+    auto gridCollapsed_1 = rewriter.create<tensor::CollapseShapeOp>(
+        loc, gridTypeCollapsed, grid_1, associations);
+    AffineExpr d0 = getAffineDimExpr(0, op->getContext());
+    AffineExpr d2 = getAffineDimExpr(2, op->getContext());
+    AffineExpr d3 = getAffineDimExpr(3, op->getContext());
+    SmallVector<AffineMap> gridMaps{
+        rewriter.getMultiDimIdentityMap(gridRank),
+        AffineMap::get(4, 0, {d0, d2, d3}, op->getContext()),
+        AffineMap::get(4, 0, {d0, d2, d3}, op->getContext())};
+    SmallVector<utils::IteratorType> gridIterators(
+        gridRank, utils::IteratorType::parallel);
+    SmallVector<int64_t> resultShape = makeShapeTorchCompatible(inputShape);
+    resultShape[2] = gridShape[1];
+    resultShape[3] = gridShape[2];
+    Value resultFinal =
+        rewriter.create<tensor::EmptyOp>(loc, resultShape, floatType);
+    auto resultFinalType = resultFinal.getType().cast<RankedTensorType>();
+    Value sGrid =
+        rewriter
+            .create<linalg::GenericOp>(
+                loc, resultFinalType,
+                ValueRange{gridCollapsed_0, gridCollapsed_1}, resultFinal,
+                gridMaps, gridIterators,
+                [&](OpBuilder &b, Location loc, ValueRange args) {
+                  Value gr_0 = args[0];
+                  Value gr_1 = args[1];
+                  Value oneFloat = b.create<arith::ConstantOp>(
```

Comment:
Probably group above with constants. Tends to clean up the IR a bit.

Reply:
yes I pushed constants to the top